### DISCUSSION TITLE
Creating CR objects

### DISCUSSION THREAD

**Discussion by TheCodeAddiction** [operator-framework/java-operator-sdk#1588] [Q&A] [answered]
Hey! I am trying to create an operator that is quite simple, but I am have some questions.

What I am trying to do is quite simple:
* I have created a operator for my CR called _foo_
* I now want my operator, to create a CR _bar_. Bar has its own stand alone operator currently deployed in our cluster.
* Currently, I don't need to implement ownership between foo and bar, however in the future, that is something I am gonna do.

Insides _foo's_ reconcile method, how do I apply a _bar_ CR to my cluster? Could someone give me a quick demo on how to deploy a CR with the operator_sdk framework?

**Comment by metacosm** [comment, marked as answer]
Take a look at: https://github.com/quarkiverse/quarkus-operator-sdk/tree/main/samples/joke. This sample operator does the same kind of thing you're trying to do: when a `JokeRequest` CR is applied to the cluster, it queries an external joke API and uses the retrieved information to create a `Joke` CR on the cluster. If you have a CRD for your `bar` CR, you could generate the Java code for it using Fabric8's client java code generator as well. Take a look at https://github.com/java-operator-sdk/java-operator-sdk/tree/main/sample-operators/leader-election which demonstrates this contract-first approach.
